src/main/java/de/tum/in/www1/artemis/repository/StatisticsRepository.java

```diff
@@ -114,4 +205,343 @@
+     */
+    private List<Map<String, Object>> fillMapList(Map<Object, List<String>> users, SpanType span, ZonedDateTime startDate) {
+        List<Map<String, Object>> returnList = new ArrayList<>();
+        users.forEach((k, v) -> {
+            Object start;
+            if (span == SpanType.DAY) {
+                start = startDate.withHour((Integer) k);
+            }
+            else if (span == SpanType.WEEK || span == SpanType.MONTH) {
+                start = startDate.withDayOfMonth((Integer) k);
+            }
+            else if (span == SpanType.QUARTER) {
+                int year = (Integer) k < getWeekOfDate(startDate) ? startDate.getYear() + 1 : startDate.getYear();
+                ZonedDateTime firstDateOfYear = ZonedDateTime.of(year, 1, 1, 0, 0, 0, 0, startDate.getZone());
+                start = getWeekOfDate(firstDateOfYear) == 1 ? firstDateOfYear.plusWeeks(((Integer) k) - 1) : firstDateOfYear.plusWeeks((Integer) k);
+            }
+            else {
+                start = startDate.withMonth(getMonthIndex((Month) k));
+            }
+            Map<String, Object> listElement = new HashMap<>();
+            listElement.put("day", start);
+            listElement.put("amount", (long) v.size());
+            returnList.add(listElement);
+        });
+        return returnList;
+    }
+
+    private Integer getMonthIndex(Month month) {
+        return switch (month) {
+            case JANUARY -> 1;
+            case FEBRUARY -> 2;
+            case MARCH -> 3;
+            case APRIL -> 4;
+            case MAY -> 5;
+            case JUNE -> 6;
+            case JULY -> 7;
+            case AUGUST -> 8;
+            case SEPTEMBER -> 9;
+            case OCTOBER -> 10;
+            case NOVEMBER -> 11;
+            case DECEMBER -> 12;
+        };
+    }
+
+    default Integer getWeekOfDate(ZonedDateTime date) {
+        LocalDate localDate = date.toLocalDate();
+        TemporalField woy = WeekFields.of(DayOfWeek.MONDAY, 4).weekOfWeekBasedYear();
+        return localDate.get(woy);
+    }
+
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType DAY
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    default Integer[] createResultArrayForDay(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        for (Map<String, Object> map : outcome) {
+            int hour = ((ZonedDateTime) map.get("day")).getHour();
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;
+            for (int i = 0; i < 24; i++) {
```

Comment:
I wonder whether there is no other mechanism to aggregate the data by the hour

Reply:
Improved the mechanism so we do not have to iterate through the possible spots